shell/key-bindings.zsh

```diff
@@ -103,8 +103,10 @@ fzf-history-widget() {
   local ret=$?
   if [ -n "$selected" ]; then
     num=$selected[1]
-    if [ -n "$num" ]; then
+    if [[ "$num" =~ ^-?[0-9]+$ ]]; then
       zle vi-fetch-history -n $num
```

Comment:
I also found that `vi-fetch-history` does not always work.
Step to reproduce: 
```
CTRL_R 
-> select entry_at_n
-> insert entry_at_n to prompt (work normally)
-> clear prompt 
-> CTRL_R again 
-> select entry_at_n
-> get empty prompt (not work anymore)
```

My guess is `vi-fetch-history` does not work properly when re-select entries that were selected before. It might also mess-up with setopt `APPEND_HISTORY`, `INC_APPEND_HISTORY` and `SHARE_HISTORY`
So i suggest to change this behavior too normal text replacement.
```suggestion
      LBUFFER="${selected:1}"
```

Reply:
The downside of LBUFFER approach is that it doesn't work nicely with multi-line commands.

```sh
echo '
  hello
  world
' &&
echo '
  foo
  bar
'
```

```
$ echo '\n hello\n world\n' &&\necho '\n foo\n bar\n'

 hello
 world

zsh: command not found: necho
```

Reply:
Got it. LBUFFER and echo can also preserve new lines, but not preserve raw input escape `\n`.
Can't find a better solution, so i'm gonna reverse it.

```sh
LBUFFER="echo ${selected:1}"
```
```sh
echo '
  hell\n\no
  world
' &&
echo '
  foo
  bar
'
```
```sh
$ echo '
  hell

o
  world
' &&
echo '
  foo
  bar
'
```